src/encoder.rs

```diff
@@ -2933,22 +2933,22 @@ fn encode_partition_topdown<T: Pixel, W: Writer>(
       (bsize > fi.partition_range.min && is_square) || must_split
     };
 
-  let mut rdo_output =
-    block_output.clone().unwrap_or(PartitionGroupParameters {
+  let mut rdo_output = if let Some(block_output) = block_output {
+    block_output.clone()
+  } else {
+    PartitionGroupParameters {
```

Comment:
How does using `unwrap_or_else()` compare? It should be equivalent after optimisation but more concise.
```suggestion
  let mut rdo_output =
    block_output.clone().unwrap_or_else(|| PartitionGroupParameters {
```

Reply:
It does seem equivalent after optimization. For some reason I thought clone and then unwrap_or_else would be worse then cloning inside an if-let block, but I guess not.